### DISCUSSION TITLE
Detector for SEI Cert ERR08-J

### DISCUSSION THREAD

**Discussion by gamesh411** [spotbugs/spotbugs#1528] [Q&A]
Hi!
I am new to contributing to Spotbugs, and I would like to implement a check for SEI Cert rule [ERR08-J](https://wiki.sei.cmu.edu/confluence/display/java/ERR08-J.+Do+not+catch+NullPointerException+or+any+of+its+ancestors) and maybe a few related rules.
I have been looking around the repo for anything related and found that [RuntimeExceptionCapture (REC)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rec-exception-is-caught-when-exception-is-not-thrown-rec-catch-exception) implemented [here](https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/RuntimeExceptionCapture.java). IMHO it is only mildly related, and would like to ask for your advice, whether I should contribute the check as a standalone detector or as an extension to REC (or some other I may have missed) detector.

**Comment by winne42** [comment]
@gamesh411 As far as I understand, you have implemented DCN_NULLPOINTER_EXCEPTION checker and it was released with SpotBugs 4.5.0?
